newsroom/search.py

```diff
@@ -259,7 +259,43 @@ def get_internal_request(self, search):
         return internal_req
 
     def set_bool_query_from_filters(self, bool_query: Dict[str, Any], filters: Dict[str, Any]):
-        bool_query["must"] += [{"terms": {self.get_aggregation_field(key): val}} for key, val in filters.items() if val]
+        for key, val in filters.items():
+            if not val:
+                continue
+            nested_config = self.get_nested_config(key)
+            if nested_config:
+                query = {
+                    "nested": {
+                        "path": nested_config["parent"],
+                        "query": {
+                            "bool": {
+                                "must": [
+                                    {
+                                        "term": {
+                                            f"{nested_config['parent']}.{nested_config['field']}": nested_config[
+                                                "value"
+                                            ]
+                                        }
+                                    },
+                                    {"terms": {f"{nested_config['parent']}.{nested_config['searchfield']}": val}},
+                                ],
+                            },
+                        },
+                    },
+                }
+            else:
+                query = {"terms": {self.get_aggregation_field(key): val}}
+            bool_query["must"].append(query)
+
+    def get_nested_config(self, field):
```

Comment:
There is similar functionality here: https://github.com/superdesk/newsroom-core/blob/release/2.3/newsroom/search_config.py#L27

Might be good to have this functionality in the one place, common to all resource types